ledger/src/snapshot_utils.rs

```diff
@@ -153,14 +247,22 @@ pub fn add_snapshot<P: AsRef<Path>>(snapshot_path: P, bank: &Bank) -> Result<()>
         snapshot_file_path,
     );
 
-    let snapshot_file = File::create(&snapshot_file_path)?;
-    // snapshot writer
-    let mut snapshot_stream = BufWriter::new(snapshot_file);
     // Create the snapshot
-    serialize_into(&mut snapshot_stream, &*bank)?;
     let mut bank_rc_serialize = Measure::start("create snapshot");
-    serialize_into(&mut snapshot_stream, &bank.rc)?;
+    let consumed_size =
+        serialize_snapshot_data_file(&snapshot_file_path, MAX_SNAPSHOT_DATA_FILE_SIZE, |stream| {
+            serialize_into(stream.by_ref(), &*bank)?;
+            serialize_into(stream.by_ref(), &bank.rc)?;
+            Ok(())
+        })?;
     bank_rc_serialize.stop();
+
+    datapoint_info!(
```

Comment:
Tidying up these `data_point_info!`s by making serialize_snapshot_data_file handle them might be a good idea?

Reply:
done!!: 51d786a44e587a9a8a690d032341dfbff836f96d